EVP_BytesToKey.js

```diff
@@ -1,39 +1,35 @@
 var createHash = require('create-hash');
-module.exports = function evp(password, salt, keyLen) {
-  keyLen = keyLen/8;
-  var ki = 0;
-  var ii = 0;
-  var key = new Buffer(keyLen);
-  var addmd = 0;
-  var md, md_buf;
-  var i;
-  while (true) {
-    md = createHash('md5');
-    if(addmd++ > 0) {
-       md.update(md_buf);
+
+module.exports = function evp (password, salt, keyLen) {
+  keyLen = keyLen / 8
+  
+  var offset = 0
+  var key = new Buffer(keyLen)
+  var buffer
+  
+  while (keyLen > 0) {
+    var hash = createHash('md5')
+    
+    if (buffer) {
+      hash.update(buffer)
     }
-    md.update(password);
-    md.update(salt);
-    md_buf = md.digest();
-    i = 0;
-    if(keyLen > 0) {
-      while(true) {
-        if(keyLen === 0) {
-          break;
-        }
-        if(i === md_buf.length) {
-          break;
-        }
-        key[ki++] = md_buf[i++];
-        keyLen--;
-       }
+    
+    hash.update(password)
+    hash.update(salt)
+    buffer = hash.digest()
+    
+    for (var i = 0; i < buffer.length; ++i) {
+      if (keyLen === 0) break
+      
+      key[offset] = buffer[i]
+      
+      keyLen--
+      offset++
     }
-   if(keyLen === 0) {
-      break;
-    }
-  }
-  for(i=0;i<md_buf.length;i++) {
-    md_buf[i] = 0;
   }
-  return key;
-};
\ No newline at end of file
+  
+  // zero the temporary buffer
+  buffer.fill(0)
```

Comment:
yeah we could leave a comment

On Wed, May 20, 2015 at 9:59 AM Daniel Cousens <notifications@github.com>
wrote:

> In EVP_BytesToKey.js
> <https://github.com/crypto-browserify/parse-asn1/pull/12#discussion_r30703967>
> :
>
> > \ No newline at end of file
> > +
> > +  // zero the temporary buffer
> > +  buffer.fill(0)
>
> Sure, I'm just trying to ensure we don't give off any false assertions,
> this could very much still be in memory.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/crypto-browserify/parse-asn1/pull/12/files#r30703967>.
>